src/Common/ORDataObject/Contact.php

```diff
@@ -40,18 +40,24 @@ class Contact extends ORDataObject
     public function __construct($id)
     {
         parent::__construct("contact");
-        $this->_id = $id;
+        $this->id = $id;
 
         if (!empty($id)) {
             $this->populate();
         }
     }
 
-    public function setPatientPid($pid)
+    public function setContactRecord($foreign_table_name, $foreign_id)
     {
         // we set our type to be patient_id and our table type here.
-        $this->foreign_table_name = 'patient_data';
-        $this->foreign_id = $pid;
+        $this->foreign_table_name = $foreign_table_name;
+        $this->foreign_id = $foreign_id;
+
+        // if entry already exists in table, then set the id
+        $id = sqlQuery("SELECT `id` FROM `contact` WHERE `foreign_table_name` = ? AND `foreign_id` = ?", [$foreign_table_name, $foreign_id])['id'] ?? null;
+        if (!empty($id)) {
+            $this->id = $id;
```

Comment:
@bradymiller 
Since the contact table is just for mapping what tables additional addresses are used and not for mapping each address entry then I don't see a recourse other than to validate if an existing mapping already exists and carry the id forward.

While this fix will work on updating an existing or adding a single new address, it will fail when adding more than one additional address when an existing contact doesn't already exist. Say new patient and adding several additional addresses on data entry.

This is because we build a array of records to save before we persist(I think an eye towards transaction). Since a contact map for the patient doesn't exist, then a new sequence is generated populating the records array(in memory) then persisted.
The contact id should be tested for and set in a property in contact service construct somehow.

I can fix this tonight if you want or we wait.

Reply:
@sjpadgett , noted that issue on multiple addresses on testing and think may have a solution (still testing it).